pkg/core/contract_listener.go

```diff
@@ -26,17 +26,19 @@ import (
 )
 
 type ContractListener struct {
-	ID        *fftypes.UUID            `ffstruct:"ContractListener" json:"id,omitempty" ffexcludeinput:"true"`
-	Interface *fftypes.FFIReference    `ffstruct:"ContractListener" json:"interface,omitempty" ffexcludeinput:"postContractAPIListeners"`
-	Namespace string                   `ffstruct:"ContractListener" json:"namespace,omitempty" ffexcludeinput:"true"`
-	Name      string                   `ffstruct:"ContractListener" json:"name,omitempty"`
-	BackendID string                   `ffstruct:"ContractListener" json:"backendId,omitempty" ffexcludeinput:"true"`
-	Location  *fftypes.JSONAny         `ffstruct:"ContractListener" json:"location,omitempty"`
-	Created   *fftypes.FFTime          `ffstruct:"ContractListener" json:"created,omitempty" ffexcludeinput:"true"`
-	Event     *FFISerializedEvent      `ffstruct:"ContractListener" json:"event,omitempty" ffexcludeinput:"postContractAPIListeners"`
-	Signature string                   `ffstruct:"ContractListener" json:"signature" ffexcludeinput:"true"`
-	Topic     string                   `ffstruct:"ContractListener" json:"topic,omitempty"`
-	Options   *ContractListenerOptions `ffstruct:"ContractListener" json:"options,omitempty"`
+	ID         *fftypes.UUID            `ffstruct:"ContractListener" json:"id,omitempty" ffexcludeinput:"true"`
+	Interface  *fftypes.FFIReference    `ffstruct:"ContractListener" json:"interface,omitempty" ffexcludeinput:"postContractAPIListeners"`
+	Namespace  string                   `ffstruct:"ContractListener" json:"namespace,omitempty" ffexcludeinput:"true"`
+	Name       string                   `ffstruct:"ContractListener" json:"name,omitempty"`
+	BackendID  string                   `ffstruct:"ContractListener" json:"backendId,omitempty" ffexcludeinput:"true"`
+	Location   *fftypes.JSONAny         `ffstruct:"ContractListener" json:"location,omitempty" ffexcludeinput:"true"`
+	Created    *fftypes.FFTime          `ffstruct:"ContractListener" json:"created,omitempty" ffexcludeinput:"true"`
+	Event      *FFISerializedEvent      `ffstruct:"ContractListener" json:"event,omitempty" ffexcludeinput:"true"`
+	Signature  string                   `ffstruct:"ContractListener" json:"signature,omitempty" ffexcludeinput:"true"`
```

Comment:
Ok - maybe some certainty in my feedback here:
- If the `signature` lives here still, it has to be the hash in the case of `len(filters) > 1`
- I think in the `len(filters) > 1` case, we need a `signature` to be a part of each `filter`

Reply:
Agreed - have done that